packages/lib-classifier/src/store/SubjectStore.js

```diff
@@ -77,19 +79,24 @@ const SubjectStore = types
           const { path, value } = patch
           if (path === '/classifications/loadingState' && value === 'posting') self.advance()
         })
-      })
+      }, { name: 'SubjectStore Classification Observer autorun' })
       addDisposer(self, classificationDisposer)
     }
 
     function onSubjectAdvance (call, next, abort) {
       const root = getRoot(self)
-      const subject = self.active
-      const shouldShowFeedback = root.feedback.isActive && root.feedback.messages.length && !root.feedback.showModal
-      if (!shouldShowFeedback && subject && subject.shouldDiscuss) {
-        const { url, newTab } = subject.shouldDiscuss
-        openTalkPage(url, newTab)
+      const validSubjectReference = isValidReference(() => self.active)
+      if (validSubjectReference) {
+        const subject = self.active
+        const shouldShowFeedback = root.feedback.isActive && root.feedback.messages.length && !root.feedback.showModal
+        if (!shouldShowFeedback && subject && subject.shouldDiscuss) {
+          const { url, newTab } = subject.shouldDiscuss
+          openTalkPage(url, newTab)
+        }
+        next(call)
+      } else {
+        next(call)
```

Comment:
Should this be `abort()`? If you always want to call `next(call)`, then it should be moved outside the if block.